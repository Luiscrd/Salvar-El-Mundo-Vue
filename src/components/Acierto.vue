<template>
    <div class="card text-center">
        <div class="card-header text-danger">Acierto</div>
        <div class="card-body text-dark d-flex flex-column align-items-center">
            <h1 class="card-title mb-4 mt-5">Bien!!, has acertado</h1>

            <h2>Sigue as√≠ y podremos salvarnos</h2>

            <a @click="next" class="btn btn-primary mt-4">Siguiente</a>
        </div>
        <div class="card-footer text-muted">{{ fecha }}</div>
    </div>
</template>

<script>
import { useRouter } from 'vue-router'
import { useStore } from 'vuex'
import { computed, ref } from 'vue'

export default {
    name: 'Acierto',
    setup() {
        const store = useStore()
        const router = useRouter()
        const preguntaActual = computed(() => store.getters['preguntaActual'])
        const fecha = ref(new Date().toUTCString())

        return {
            next: () => {
                if (preguntaActual.value === 11) {
                    router.push({ name: 'fin' })

                } else {
                    router.push({ name: 'answer', params: {id: preguntaActual.value}})
                }
            },
            fecha
        }
    }
}
</script>

<style lang="scss" scoped>
.form-check-input:checked {
    background-color: #341948;
    border-color: #341948;
}

.card{
    height: 377px;
}
</style>