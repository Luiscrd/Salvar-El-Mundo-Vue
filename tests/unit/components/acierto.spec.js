import { shallowMount } from '@vue/test-utils'
import Acierto from '@/components/Acierto.vue'

describe('Formulario View', () => {

    let wrapper;

    beforeEach(() => {
          
          wrapper = shallowMount(Acierto)

    });

    test('Debe de hacer mach con el snapshot', () => {

        expect(wrapper.html()).toMatchSnapshot()

    })

    test('El texto a mostrar en el h1 debe ser "Bien!!, has acertado"', () => {

        const h1 = wrapper.find('h1')

        expect(h1.text()).toBe('Bien!!, has acertado')

    })

    test('El texto a mostrar en el h2 debe ser "Sigue así y podremos salvarnos"', () => {

        const h2 = wrapper.find('h2')

        expect(h2.text()).toBe('Sigue así y podremos salvarnos')

    })

})