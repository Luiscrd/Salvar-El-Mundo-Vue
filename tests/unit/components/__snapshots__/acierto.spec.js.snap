// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Formulario View Debe de hacer mach con el snapshot 1`] = `
<div class="card text-center">
  <div class="card-header text-danger">Acierto</div>
  <div class="card-body text-dark d-flex flex-column align-items-center">
    <h1 class="card-title mb-4 mt-5">Bien!!, has acertado</h1>
    <h2>Sigue as√≠ y podremos salvarnos</h2><a class="btn btn-primary mt-4">Siguiente</a>
  </div>
  <div class="card-footer text-muted">Tue, 08 Nov 2022 00:28:42 GMT</div>
</div>
`;
